### ISSUE TITLE
Use Python package structure

### ISSUE THREAD

**Issue by edbennett** [LupoA/rhos#27]
It would be really good to restructure this into a Python package, such that anyone can install it with `pip`, and so that you don't have to mess around with adding directories to the `PATH` to be able to import things correctly.

Python provides [a tutorial on packaging Python projects](https://packaging.python.org/en/latest/tutorials/packaging-projects/). (Note if you're skim-reading that `YOUR_USERNAME_HERE` is _not_ an essential part of a package name, that's just so you can follow the tutorial without clashing with anyone else's tutorial package.)

Essential aspects of this:

 - Create a `pyproject.toml` specifying requirements and metadata
 - Use e.g. `from .core` to import from a module in the same directory
 - Make `utils` part of `rhos`. Is `correlator` used at all? If so, move it into `rhos`, otherwise remove it.
 - Consider whether `rhos` is the best name for the package; make the directory name match
 - Add an `__init__.py` file
 - Remove `importall.py`

Once this is done, #26 will probably be easier. (If #26 is already done, then this will break CI and it will need to be fixed.)

**Comment by nickforce989** [comment]
I have started addressing this issue (togheter with issues #23 and #24, LICENSE and README) in the branch feature/package. This branch starts from Alessandro's modifications in feature/GP.

In this branch, I will also try to merge the fitting procedure of spectral densities and the usage of different kernels. Moreover, I will add a couple of things that are missing currently in the most recent version of the code.

**Comment by LupoA** [comment]
excellent, thanks! I'll assign you to this issue

**Comment by edbennett** [comment]
If HLTRho is going to be the public-facing name for the package, it would probably be a good idea to use something similar (e.g. `hltrho`) in Python rather than `rhos`, which may clash with other differently-named packages (and confuse users).

**Comment by LupoA** [comment]
if anyone is good with acronyms and shit like that I am down for a completely new name

**Comment by LupoA** [comment]
[I think we are covered with the music regardless](https://www.youtube.com/watch?v=On1AvOUjgbI)

**Comment by edbennett** [comment]
Resolved by #41.
